### PR TITLE
Encrypt messages: keys exchange, part 1

### PR DESCRIPTION
**Pull request by rabbitonweb**
This PR is first part of public key exchange.  Keys are being generated (not yet connected to `crypto` module) and stored on `initiator` side (storing on `remote` side will handled in part 2)

This part refactors a bunch of code, enabling `Kvs` effect so once we receive a public key (when `connect` is called) we can store that key in KeyValueStore.

In order for remote to store incoming public key, I need to refactor `handleEncryptionHandshake` so that it returns effect (so it also gets access to `Kvs` effect) - effectively refactoring the whole dispatcher trait. <--- this will be part of `Encrypt messages: keys exchange, part 2` as this PR is already too big

### CODE REVIEW

File: comm/src/main/protobuf/coop/rchain/comm/protocol/p2p.proto [hunk 1]
```diff
@@ -17,12 +17,15 @@ message Hello {
...
-  bytes  hello     = 2;
 }
 
 message EncryptionHandshakeResponse {
   bytes  publicKey     = 1;
-  bytes  helloBack     = 2;
+}
+
+message Frame {
+  bytes nounce   = 1;
```
Comment (kirkwood) [inline comm/src/main/protobuf/coop/rchain/comm/protocol/p2p.proto]: I think "nonce" is what we're after :)

**Comment by kirkwood** [comment]
What's the convention with `ApplicativeError_.scala`-type filenames?

**Comment by rabbitonweb** [comment]
@kirkwood this PR is ready for review

File: comm/src/main/protobuf/coop/rchain/comm/protocol/p2p.proto (hunk 1 above)
Comment (rabbitonweb) [reply comm/src/main/protobuf/coop/rchain/comm/protocol/p2p.proto]: could be :)

**Comment by rabbitonweb** [comment]
`ApplicativeError_` is introduced because the way MTL is introduced in cats (copy-pasted from scalaz 7.x) is broken. This is a workaround/hack.

**Comment by rabbitonweb** [comment]
Also I've added small change to scalafmt but i refactored changed that will be applied to modules other then `comm`

**Comment by rabbitonweb** [comment]
@kirkwood `ApplicativeError` from cats is broken - almost all MTL classes are (can deliver details if needed) thus `ApplicativeError_` is a hack. The idea here (in a nutshell) is to:
1. provide something that will act like `ApplicativeError` but not being (extending) `Applicative`)
2. implicit conversion: if you have `ApplicativeError` instance, you also have instance for `ApplicativeError_`

**Pull request merged**